pkg/cluster/acsk/action/action_create_nodepool.go

```diff
@@ -0,0 +1,183 @@
+	defer close(instanceIdsChan)
+
+	for _, nodePool := range a.nodePools {
+		go func(nodePool *model.ACSKNodePoolModel) {
+			scalingGroupRequest := ess.CreateCreateScalingGroupRequest()
+			scalingGroupRequest.SetScheme(requests.HTTPS)
+			scalingGroupRequest.SetDomain("ess." + cluster.RegionID + ".aliyuncs.com")
+			scalingGroupRequest.SetContentType(requests.Json)
+
+			a.log.WithFields(logrus.Fields{
+				"region":        cluster.RegionID,
+				"zone":          cluster.ZoneID,
+				"instance_type": nodePool.InstanceType,
+			}).Info("creating scaling group")
+
+			scalingGroupRequest.MinSize = requests.NewInteger(nodePool.MinCount)
+			scalingGroupRequest.MaxSize = requests.NewInteger(nodePool.MaxCount)
+			scalingGroupRequest.VSwitchId = cluster.VSwitchID
+			scalingGroupRequest.ScalingGroupName = fmt.Sprintf("asg-%s-%s", nodePool.Name, cluster.ClusterID)
+
+			createScalingGroupResponse, err := a.context.ESSClient.CreateScalingGroup(scalingGroupRequest)
+			if err != nil {
+				errChan <- emperror.WrapWith(err, "could not create Scaling Group", "nodePoolName", nodePool.Name, "clusterName", cluster.Name)
+				instanceIdsChan <- nil
+				return
+			}
+
+			nodePool.AsgId = createScalingGroupResponse.ScalingGroupId
+			a.log.Infof("Scaling Group with id %s successfully created", nodePool.AsgId)
+			a.log.Infof("Creating scaling configuration for group %s", nodePool.AsgId)
+
+			scalingConfigurationRequest := ess.CreateCreateScalingConfigurationRequest()
+			scalingConfigurationRequest.SetScheme(requests.HTTPS)
+			scalingConfigurationRequest.SetDomain("ess." + cluster.RegionID + ".aliyuncs.com")
+			scalingConfigurationRequest.SetContentType(requests.Json)
+
+			scalingConfigurationRequest.ScalingGroupId = nodePool.AsgId
+			scalingConfigurationRequest.SecurityGroupId = cluster.SecurityGroupID
+			scalingConfigurationRequest.KeyPairName = cluster.Name
+			scalingConfigurationRequest.InstanceType = nodePool.InstanceType
+			scalingConfigurationRequest.SystemDiskCategory = "cloud_efficiency"
+			scalingConfigurationRequest.ImageId = "centos_7_04_64_20G_alibase_20180419.vhd"
+			scalingConfigurationRequest.Tags =
+				fmt.Sprintf(`{"pipeline-created":"true","pipeline-cluster":"%s","pipeline-nodepool":"%s"`,
+					cluster.Name, nodePool.Name)
+
+			createConfigurationResponse, err := a.context.ESSClient.CreateScalingConfiguration(scalingConfigurationRequest)
+			if err != nil {
+				errChan <- emperror.WrapWith(err, "could not create Scaling Configuration", "nodePoolName", nodePool.Name, "scalingGroupId", nodePool.AsgId, "clusterName", cluster.Name)
+				instanceIdsChan <- nil
+				return
+			}
+
+			nodePool.ScalingConfId = createConfigurationResponse.ScalingConfigurationId
+
+			a.log.Infof("Scaling Configuration successfully created for group %s", nodePool.AsgId)
+
+			enableSGRequest := ess.CreateEnableScalingGroupRequest()
+			enableSGRequest.SetScheme(requests.HTTPS)
+			enableSGRequest.SetDomain("ess." + cluster.RegionID + ".aliyuncs.com")
```

Comment:
The api endpoint could be extracted into a const similar to `ossEndpointFmt = "https://oss-%s.aliyuncs.com"`